stablehlo/conversions/linalg/transforms/StablehloLegalizeToLinalg.cpp

```diff
@@ -1047,7 +1046,7 @@ struct ReshapeOpConverter final
     Value operand = adaptor.getOperand();
     auto operandType = llvm::cast<ShapedType>(operand.getType());
     Type elemType = operandType.getElementType();
-    auto resultType = llvm::cast<ShapedType>(reshapeOp.getType());
+    ShapedType resultType = reshapeOp.getType();
```

Comment:
Could we use auto everywhere? (This goes against the rule of using auto sparingly but the file uses it quite consistently).

Reply:
Good catch! Unfortunately there is an assignment below (on line 1061) that has to use `ShapedType`, so this one can't use `auto`.